assets/javascripts/discourse/controllers/group-reports-show.js

```diff
@@ -43,34 +44,41 @@ export default class GroupReportsShowController extends Controller {
       : "query-group-bookmark";
   }
 
-  @action
-  run() {
+  @bind
+  async run() {
     this.loading = true;
     this.showResults = false;
 
-    ajax(`/g/${this.get("group.name")}/reports/${this.model.id}/run`, {
-      type: "POST",
-      data: {
-        params: JSON.stringify(this.model.params),
-        explain: this.explain,
-      },
-    })
-      .then((result) => {
-        this.results = result;
-        if (!result.success) {
-          return;
+    try {
+      const response = await ajax(
+        `/g/${this.get("group.name")}/reports/${this.model.id}/run`,
+        {
+          type: "POST",
+          data: {
+            params: JSON.stringify(this.model.params),
+            explain: this.explain,
+          },
         }
+      );
 
-        this.showResults = true;
-      })
-      .catch((err) => {
-        if (err.jqXHR && err.jqXHR.status === 422 && err.jqXHR.responseJSON) {
-          this.results = err.jqXHR.responseJSON;
-        } else {
-          popupAjaxError(err);
-        }
-      })
-      .finally(() => (this.loading = false));
+      this.results = response;
+      if (!response.success) {
```

Comment:
This feels like a code smell (was existing logic).. if the response is not successful wouldn't we leave the user in a permanent `loading` state since we never make it to setting the loading to false?

Reply:
Yeah, `this.loading = false` should probably be moved to a `finally` block